### DISCUSSION TITLE
BFV Plaintext modulu

### DISCUSSION THREAD

**Discussion by czh-rot** [tuneinsight/lattigo#444] [Q&A]
I'm thinking about two things:
1. For the plaintext module of the BFV scheme, suppose I start with T=2^16+1 and encrypt a vector. Can I change this ciphertext to the encryption under the T=2^32+1 by doing something like modulus switch?
2.  Whether the ciphertext encrypted by different T can not be directly added or multiplied?

**Comment by czh-rot** [comment]
Let me explain in more details. I've made a try and the result proves that the Q2, i.e., different ciphertexts generated by different Parameter Sets can not be directly computed.
Assume that I use the Parameter Set(N, Q, T) = {2^16+1, 880bit, 2^15} to encrypt a vector to get the ciphertext c0. I want convert the c0 to c1 (c1 is the encryption under the Parameter Set(N, Q, T) = {2^32+1, 880bit, 2^15}). What should I do?

**Reply by Pro7ech** [reply]
1) Homomorphically decode the plaintext by evaluating a homomorphic NTT modulo 2^{16} + 1
2) Homomorphically encode the plaintext by evaluating a homomorphic INTT modulo 2^{32} + 1

This will return plaintext modulo 2^{32} +1 with values scaled by (2^{32}+1)/(2^{16}+1)

Unfortunately Lattigo currently doesn't support such operation for BFV/BGV.
